src/storage/patch/RdfImmutableCheckPatcher.ts

```diff
@@ -0,0 +1,75 @@
+import type { Quad } from 'rdf-js';
+import type { AuxiliaryStrategy } from '../../http/auxiliary/AuxiliaryStrategy';
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { ConflictHttpError } from '../../util/errors/ConflictHttpError';
+import { serializeQuads, uniqueQuads } from '../../util/QuadUtil';
+import { cloneRepresentation } from '../../util/ResourceUtil';
+import { readableToQuads, readableToString } from '../../util/StreamUtil';
+import { RDF, PIM, LDP } from '../../util/Vocabularies';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+export class RdfImmutableCheckPatcher extends RepresentationPatcher {
```

Comment:
```suggestion
export class ImmutableMetadataPatcher extends RepresentationPatcher {
```
Makes more sense I think? Since it's specifically written to prevent certain metadata changes.

Also missing TSdoc above the class definition.